drivers/SmartThings/matter-appliance/src/matter-oven/init.lua

```diff
@@ -0,0 +1,313 @@
+    if val < OVEN_MIN_TEMP_IN_C or val > OVEN_MAX_TEMP_IN_C then
+      if limit_field == setpoint_limit_device_field.MIN_TEMP then
+        val = OVEN_MIN_TEMP_IN_C
+      else
+        val = OVEN_MAX_TEMP_IN_C
+      end
+    end
+
+    log.info("Setting " .. field .. " to " .. string.format("%s", val))
+    device:set_field(field, val, { persist = true })
+  end
+end
+
+-- Capability Handlers --
+local function handle_oven_mode(driver, device, cmd)
+  log.info(string.format("handle_oven_mode mode: %s", cmd.args.mode))
+  local ep = component_to_endpoint(device, cmd.component)
+  local supportedOvenModesMap = device:get_field(SUPPORTED_OVEN_MODES_MAP) or {}
+  local supportedOvenModes = supportedOvenModesMap[string.format(ep)] or {}
+  for i, mode in ipairs(supportedOvenModes) do
+    if cmd.args.mode == mode then
+      device:send(clusters.OvenMode.commands.ChangeToMode(device, ep, i - 1))
+      return
+    end
+  end
+  log.warn("handle_oven_mode received unsupported mode: ".." for endpoint: "..ep)
+end
+
+local function handle_temperature_setpoint(driver, device, cmd)
+  local tn_eps = embedded_cluster_utils.get_endpoints(device, clusters.TemperatureControl.ID,
+    { feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER })
+  if #tn_eps == 0 then
+    device.log.warn(string.format("Device does not support TEMPERATURE_NUMBER feature"))
+    return
+  end
+  device.log.info(string.format("handle_temperature_setpoint: %s", cmd.args.setpoint))
+
+  local value = cmd.args.setpoint
+  local _, temp_setpoint = device:get_latest_state(
+    cmd.component, capabilities.temperatureSetpoint.ID,
+    capabilities.temperatureSetpoint.temperatureSetpoint.NAME,
+    0, { value = 0, unit = "C" }
+  )
+  local ep = component_to_endpoint(device, cmd.component)
+  local min_field = string.format("%s-%d", setpoint_limit_device_field.MIN_TEMP, ep)
+  local max_field = string.format("%s-%d", setpoint_limit_device_field.MAX_TEMP, ep)
+  local min = device:get_field(min_field) or OVEN_MIN_TEMP_IN_C
+  local max = device:get_field(max_field) or OVEN_MAX_TEMP_IN_C
+
+  -- For RPC >= 5, temperature capabilities emitted from the driver are converted
+  -- to local temperature scale by the hub, so the driver should always use Celsius.
+  -- For lower versions, we can emit the capabilities with Fahrenheight as the hub
+  -- does not do this conversion.
+  if version.rpc >= 5 then
+    if value > OVEN_MAX_TEMP_IN_C then
+      value = utils.f_to_c(value)
+    end
+  else
+    local unit = "C"
+    if value <= OVEN_MAX_TEMP_IN_C then
+      device:set_field(CURRENT_CONFIGURED_UNIT, unit, { persist = true })
+      if temp_setpoint.unit == "F" then
+        temp_setpoint.value = utils.f_to_c(temp_setpoint.value)
+        temp_setpoint.unit = unit
+      end
+    else
+      unit = "F"
+      device:set_field(CURRENT_CONFIGURED_UNIT, unit, { persist = true })
+      value = utils.f_to_c(value)
```

Comment:
In this case we have identified that the new temperature setpoint is in F, but we're converting it back to C. That's correct for RCP version >=5 but I think for older hubs we should not do this, is that correct?

Reply:
I think this particular `value` variable is only being used for the device cluster command (`clusters.TemperatureControl.commands.SetTemperature()`) where the unit needs to be in C. 

The one for capability temp set point is being set to a value in F with the unit update if its unit is C in this arm. Am I missing something?

Reply:
Yes that is correct @samadDotDev . The cluster command is always in C and the capability can be in either unit.

Reply:
This logic is going to have to be deprecated, per Nick's message [here](https://smartthings.slack.com/archives/C07KHPPP8H5/p1727462113093399). Resolving for this reason.